src/controller/CHIPDeviceController.cpp

```diff
@@ -682,12 +683,13 @@ CHIP_ERROR DeviceCommissioner::Shutdown()
     ChipLogDetail(Controller, "Shutting down the commissioner");
 
     // Check to see if pairing in progress before shutting down
-    CommissioneeDeviceProxy * device = mDeviceBeingCommissioned;
+    CommissioneeDeviceProxy * device = mDeviceInProcessOfPase;
     if (device != nullptr && device->IsSessionSetupInProgress())
     {
         ChipLogDetail(Controller, "Setup in progress, stopping setup before shutting down");
         OnSessionEstablishmentError(CHIP_ERROR_CONNECTION_ABORTED);
     }
+    // TODO: If we have a commissioning step in progress, is there a way to cancel that callback?
```

Comment:
https://github.com/project-chip/connectedhomeip/pull/16127 added a release of the pool to the shutdown function. This PR is just on an older base line.

Reply:
> All of the commissioning steps are IM interactions over pase so I'm not sure what you mean here.

I mean that if we're still in the `IsSessionSetupInProgress()` stage then we are good.

If we are into the IM bits, then just doing the pool release is not enough, afaict, because that does not remove the PASE session from the session manager, so we can still get callbacks to deleted objects.  As far as I can tell.  But maybe I'm missing something?

Reply:
Nope -  I think we're on the same page. The todo here is to shut down any pending cluster callbacks - I'm not sure how yet, so I just stuck this here. This used to be handled by the commissioner because we had cancellable callbacks, but that's not possible now, so I'll need to dig deeper.

The release from the pool does call the destructor on the CommissioneeDeviceProxy, but that does nothing currently. The commissioner does not have direct access to the pase session, so removing the session SHOULD happen in the proxy, as far as I can tell. But I need to sort out how the proxies relate to the pairing and session info to ensure I'm not plowing over a case session when tearing down pase.

Reply:
Handling this in a followup makes total sense to me.

Fwiw, just in terms of how to clean up the PASE session, I think `ExpirePairing` on the SessionManager will do the trick.  That can be done by the CommissioneeDeviceProxy or the commissioner itself (since it can get the SessionHandle from the proxy).  Doing it from the CommissioneeDeviceProxy would make a lot of sense.

Reply:
Tracking in an issue here: https://github.com/project-chip/connectedhomeip/issues/16440